corehq/apps/app_manager/models.py

```diff
@@ -4518,7 +4524,10 @@ def make_build(self, comment=None, user_id=None, previous_version=None):
             # but check explicitly so as not to change the _id if it exists
             copy._id = copy.get_db().server.next_uuid()
 
-        copy.set_form_versions(previous_version)
+        force_new_forms = False
+        if self.build_profiles != previous_version.build_profiles:
+            force_new_forms = True
+        copy.set_form_versions(previous_version, force_new_forms)
```

Comment:
this bumps version numbers of the forms whenever build profiles change to force the app to update forms. otherwise if a build profile was changed (by adding a language for example) or deleted, the app wasnt updating the form and it would not get the new languages